src/util.jl

```diff
@@ -336,4 +341,117 @@ function shiftin!(a::AbstractVector{T}, b::AbstractVector{T}) where T
     return a
 end
 
+# DELAY FINDING UTILITIES
+
+"""
+    finddelay(x, y)
+
+Estimate the delay of x with respect to y by locating the peak of their
+cross-correlation.
+
+The output delay will be positive when x is delayed with respect y, negative if
+advanced, 0 otherwise.
+
+# Example
+```jldoctest
+julia> finddelay([0, 0, 1, 2, 3], [1, 2, 3])
+2
+
+julia> finddelay([1, 2, 3], [0, 0, 1, 2, 3])
+-2
+```
+"""
+function finddelay(x::AbstractVector{<: Real}, y::AbstractVector{<: Real})
```

Comment:
@galenlynch the function below is searching for the peak of the absolute correlation.

```julia
s = xcorr(y, x)
max_corr = maximum(abs, s)
max_idxs = Compat.findall(x -> abs(x) == max_corr, s)
```